### PR TITLE
Changed configurationPid prefixes from 'org.openhab.core' ot 'org.openhab'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Changed configurationPid prefixes from 'org.openhab.core' ot 'org.openhab'
- Minor improvements for integration tests using configurationPids

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.io.rest/src/main/java/org/openhab/core/io/rest/internal/filter/ProxyFilter.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
  */
 @Provider
 @PreMatching
-@Component(configurationPid = "org.openhab.core.io.rest.proxyfilter", immediate = true, service = ProxyFilter.class)
+@Component(configurationPid = "org.openhab.io.rest.proxyfilter", immediate = true, service = ProxyFilter.class)
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.io.rest/src/main/java/org/openhab/core/io/rest/internal/filter/ProxyFilter.java]:
```suggestion
@Component(configurationPid = "org.openhab.proxyfilter", immediate = true, service = ProxyFilter.class)
```

File: bundles/org.openhab.core.io.rest/src/main/java/org/openhab/core/io/rest/internal/resources/RootResource.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
  * @author Kai Kreuzer - Initial contribution
  */
 @Path("/")
-@Component(service = RootResource.class, configurationPid = "org.openhab.core.io.rest.root")
+@Component(service = RootResource.class, configurationPid = "org.openhab.io.rest.root")
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.io.rest/src/main/java/org/openhab/core/io/rest/internal/resources/RootResource.java]:
```suggestion
@Component(service = RootResource.class, configurationPid = "org.openhab.restroot")
```

File: bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicMultiServiceMultiActions.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
  *
  * @author Stefan Triller - Initial contribution
  */
-@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, configurationPid = "org.openhab.core.MagicMultiAction")
+@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, configurationPid = "org.openhab.MagicMultiAction")
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicMultiServiceMultiActions.java]: no camel case
```suggestion
@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, configurationPid = "org.openhab.magicmultiaction")
```

File: bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicSingleActionService.java [hunk 4]
```diff
@@ -33,7 +33,7 @@
  *
  * @author Stefan Triller - Initial contribution
  */
-@Component(configurationPid = "org.openhab.core.automation.action.magicSingleActionService", property = {
+@Component(configurationPid = "org.openhab.automation.action.magicSingleActionService", property = {
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicSingleActionService.java]:
```suggestion
@Component(configurationPid = "org.openhab.magicsingleactionservice", property = {
```

File: bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/service/MagicMultiInstanceService.java [hunk 5]
```diff
@@ -27,7 +27,7 @@
  *
  * @author Stefan Triller - Initial contribution
  */
-@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, service = MagicMultiInstanceService.class, configurationPid = "org.openhab.core.magicMultiInstance")
+@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, service = MagicMultiInstanceService.class, configurationPid = "org.openhab.magicMultiInstance")
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/service/MagicMultiInstanceService.java]:
```suggestion
@Component(immediate = true, configurationPolicy = ConfigurationPolicy.REQUIRE, service = MagicMultiInstanceService.class, configurationPid = "org.openhab.magicmultiinstance")
```

File: bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicSingleActionService.java (hunk 4 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.test.magic/src/main/java/org/openhab/core/magic/binding/internal/automation/modules/MagicSingleActionService.java]: I changes this one to: `"org.openhab.magicsingleaction"`

**Pull request merged**